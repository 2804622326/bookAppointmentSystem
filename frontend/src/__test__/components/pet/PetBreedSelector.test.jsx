import React from 'react';
import { render, waitFor, fireEvent, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import PetBreedSelector from '../../../components/pet/PetBreedSelector.jsx';

jest.mock('../../../components/pet/PetService.js', () => ({
  getAllPetBreeds: jest.fn(() => Promise.resolve({ data: ['Bulldog'] }))
}));

test('loads breeds and handles change', async () => {
  const onChange = jest.fn();
  const { asFragment } = render(<PetBreedSelector petType='Dog' value='' onChange={onChange} />);
  await waitFor(() => screen.getByText('Bulldog'));
  fireEvent.change(screen.getByRole('combobox'), { target: { value: 'Bulldog' } });
  expect(onChange).toHaveBeenCalled();
  expect(asFragment()).toMatchSnapshot();
});
