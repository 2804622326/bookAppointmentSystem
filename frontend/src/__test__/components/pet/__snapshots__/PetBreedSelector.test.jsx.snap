// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loads breeds and handles change 1`] = `
<DocumentFragment>
  <div
    class="col"
  >
    <select
      class="form-control"
      id="petBreed"
      name="breed"
      required=""
    >
      <option
        value=""
      >
        ...select breed...
      </option>
      <option
        value="Bulldog"
      >
        Bulldog
      </option>
      <option
        value="add-new-item"
      >
        Add New
      </option>
    </select>
  </div>
</DocumentFragment>
`;
